app/controllers/vacancies_controller.rb

```diff
@@ -1,25 +1,76 @@
 class VacanciesController < ApplicationController
   def index
-    @filters = VacancyFilters.new(params)
+    @filters = VacancyFilters.new(search_params)
     @sort = VacancySort.new(default_column: 'expires_on', default_order: 'asc')
                        .update(column: sort_column, order: sort_order)
     @query = Vacancy.public_search(filters: @filters, sort: @sort)
     records = @query.records
 
     @vacancy_count = records.count
-    @vacancies = VacanciesPresenter.new(records.page(params[:page]))
+    @vacancies = VacanciesPresenter.new(records.page(page))
   end
 
   def show
-    vacancy = Vacancy.published.friendly.find(params[:id])
+    vacancy = Vacancy.published.friendly.find(id)
     @vacancy = VacancyPresenter.new(vacancy)
   end
 
-  def sort_column
+  private def search_params
+    params.permit(:keyword, :location, :minimum_salary, :maximum_salary, :phase, :phase, :working_pattern).to_hash
+  end
+
+  def params
+    sanitised_params = super.each_pair do |key, value|
+      super[key] = Sanitize.fragment(value)
+    end
+    ActionController::Parameters.new(sanitised_params)
+  end
+
```

Comment:
I do prefer the repetition of the `private` prefix over the block because it becomes trivially easy to accidentally make things public with the wrong ordering of the file. That said I don't feel that strongly so I've switched it over :)

Reply:
@tahb thank you. Approved.